### DISCUSSION TITLE
Any tool to export data from cortex to RDBMS or file system

### DISCUSSION THREAD

**Discussion by ChinmayMishra** [cortexproject/cortex#3732] [Q&A]
The HTTP API can be used to query data from cortex and push it to any other system. Is anything already available to achieve this?

**Comment by pstibrany** [comment]
Cortex supports Prometheus-compatible API for querying data (https://prometheus.io/docs/prometheus/latest/querying/api/#expression-queries) and for remote-read (https://github.com/prometheus/prometheus/blob/master/prompb/remote.proto).

If you use blocks engine in Cortex and you have access to cloud storage where Cortex uploads blocks, then blocks created by Cortex are basically just Prometheus TSDB blocks, and there are some tools for working with them (eg. promtool, part of Prometheus).

**Reply by ChinmayMishra** [reply]
Thanks @pstibrany for a quick replay. If you can point me to some working examples that will be a great help.

**Comment by jtlisi** [comment]
@ChinmayMishra You may be able to make use of some tooling in the Thanos community: https://github.com/thanos-community/obslytics

This is a tool that lets you retrieve data using the Prometheus remote-read API or directly from the store-gateway services. It allows for the data to be exported in the Parquet data format.

**Reply by ChinmayMishra** [reply]
Thanks @jtlisi . Let me try this.
